include/boost/math/fft/fftw_backend.hpp

```diff
@@ -574,28 +574,14 @@
     // -> size(out) >= N
     {
       const std::size_t N = size();
-      vector_t<real_value_type> tmp(out,out+N);
```

Comment:
The algorithm that we use for rfft for powers of 2, comes naturally in-place if we use the halfcomplex representation used by FFTW.